src/universal/components/UserHub/UserHub.js

```diff
@@ -17,23 +19,74 @@ const UserHub = (props) => {
     dispatch(push('/me/settings'));
   };
 
+  const onCloseSettingsClick = (event) => {
+    const {dispatch} = props;
+    event.preventDefault();
+    dispatch(push('/me'));
+  };
+
+  const onSignOutClick = (event) => {
+    const {dispatch} = props;
+    event.preventDefault();
+    dispatch(push('/logout'));
+  };
+
   return (
-    <div className={styles.root}>
-      <img alt="Me" className={styles.avatar} src={avatar} />
-      <div className={styles.info}>
-        <div className={styles.name}>{preferredName}</div>
-        <div className={styles.email}>{email}</div>
-      </div>
-      <div className={styles.settings}>
-        <div className={styles.settingsIcon} title="My Preferences">
-          <FontAwesome name="cog" onClick={(e) => onSettingsClick(e)} />
+    <div>
+      {props.activeArea !== 'settings' ?
+        <div className={styles.root}>
+          <img alt="Me" className={styles.avatar} src={avatar} />
+          <div className={styles.info}>
+            <div className={styles.name}>{preferredName}</div>
+            <div className={styles.email}>{email}</div>
+          </div>
+          <div className={styles.settings}>
+            <div className={styles.settingsIcon} title="My Preferences">
+              <FontAwesome
+                name="cog"
+                onClick={(e) => onSettingsClick(e)}
```

Comment:
shouldn't need a thunk for this. just passing in the handler will be OK. will PR